piff/gp_interp_2pcf.py

```diff
@@ -24,10 +24,144 @@
 from sklearn.neighbors import KNeighborsRegressor
 from scipy import optimize
 from scipy.linalg import cholesky, cho_solve
+from scipy.stats import binned_statistic_2d
+import itertools
 
 from .interp import Interp
 from .star import Star, StarFit
 
+class bootstrap_2pcf(object):
+
+    def __init__(self, X, y, y_err,
+                 MIN=None, MAX=None, nbins=20, anisotropy=False):
+        """Fit statistical uncertaintie on two-point correlation function using bootstraping.
+
+        :param X:           The independent covariates.  (n_samples, 2)
+        :param y:           The dependent responses.  (n_samples, n_targets)
+        :param y_err:       Error of y. (n_samples, n_targets)
+        :param MIN:         Minimum bin for treecorr. (float)
+        :param MAX:         Maximum bin for treecorr. (float)
+        :param anisotropy:  2D 2-point correlation function.
+                            Used 2D correlation function for the
+                            fiting part of the GP. (Boolean)
+        """
+        self.X = X
+        self.y = y
+        self.y_err = y_err
+
+        size_x = np.max(self.X[:,0]) - np.min(self.X[:,0])
+        size_y = np.max(self.X[:,1]) - np.min(self.X[:,1])
+        rho = float(len(self.X[:,0])) / (size_x * size_y)
+        if MIN is None:
+            MIN = np.sqrt(1./rho)
```

Comment:
Also, I think it might be ok to use min_sep=0 in the TwoD case.  I make sure that the pairs that are really the same point twice don't get counted as "pairs", even if min_sep=0.   So it's not obvious to me that there is a need to cut off the calculation at some minimum > 0.